core/src/main/scala/codesearch/core/util/Helper.scala

```diff
@@ -55,6 +57,13 @@ object Helper {
     }
   }
 
+  def extractFile(path: String)(implicit ec: ExecutionContext): Future[Option[List[String]]] =
+    Future {
+      Try {
+        Source.fromFile(path, "UTF-8").getLines
```

Comment:
`scala.io.Source` uses blocking IO, so `extractFile` should be called with special ExecutionContext, if you want to use it with Future